docker/irods_clients_dev/ubuntu/Dockerfile

```diff
@@ -0,0 +1,64 @@
+ARG BASE_IMAGE=ubuntu:bionic
+FROM $BASE_IMAGE
+
+# Other versions available on bionic are 4.3.0
+ARG IRODS_VERSION="4.2.11"
+
+RUN echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections && \
+    apt-get update && \
+    apt-get install -q -y --no-install-recommends \
+    apt-utils \
+    ca-certificates \
+    curl \
+    dirmngr \
+    gpg \
+    gpg-agent \
+    lsb-release \
+    locales && \
+    locale-gen en_GB en_GB.UTF-8 && \
+    localedef -i en_GB -c -f UTF-8 -A /usr/share/locale/locale.alias en_GB.UTF-8
+
+ENV LANG=en_GB.UTF-8 \
+    LANGUAGE=en_GB \
+    LC_ALL=en_GB.UTF-8
+
+RUN curl -sSL https://packages.irods.org/irods-signing-key.asc | apt-key add - && \
+    echo "deb [arch=amd64] https://packages.irods.org/apt/ $(lsb_release -sc) main" |\
+    tee /etc/apt/sources.list.d/renci-irods.list && \
+    apt-get update && \
+    apt-get install -q -y --no-install-recommends \
+    irods-dev="${IRODS_VERSION}-1~$(lsb_release -sc)" \
+    irods-runtime="${IRODS_VERSION}-1~$(lsb_release -sc)" \
+    irods-icommands="${IRODS_VERSION}-1~$(lsb_release -sc)"
+
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys "E1DD270288B4E6030699E45FA1715D88E1DF1F24" && \
```

Comment:
I think apt-key may be deprecated. And you are using the curl + apt-key method in other places so maybe better to use the same everywhere if possible. The debian people have page with an explanation of how they want people to do it https://wiki.debian.org/DebianRepository/UseThirdParty so essentially download the key and dearmor save in specific path. I think the idea is that you can rm the file when you are done installing the tools you want instead of leaving it in the chain and trust thereafter.

Reply:
I attempted to follow the best practice described, but it failed for me on Ubuntu bionic and wasn't quite clear to me whether I was doing something wrong, or that it's not expected to work on an older distro like that.

I've opened https://github.com/wtsi-npg/containers/issues/49 to address this across all the Dockerfiles as a new enhancement.